types/address.go

```diff
@@ -571,6 +576,85 @@ func (ca ConsAddress) Format(s fmt.State, verb rune) {
 	}
 }
 
+// ----------------------------------------------------------------------------
+// EE address
+// ----------------------------------------------------------------------------
+//
+
+// EEAddress length is 32
+type EEAddress []byte
+
+func (ee EEAddress) Bytes() []byte {
+	return ee
+}
+
+func verifyEEAddress(src []byte) error {
+	if len(src) != 32 {
+		return errors.New("Incorrect address length")
+	}
+
+	return nil
+}
+
+// GetEEAddressFromCryptoPubkey converts blake2b 32 byte address from amino-encoded 33 bit secp256k1 public key
+func GetEEAddressFromCryptoPubkey(cryptoPubKey crypto.PubKey) (EEAddress, error) {
+	cdc := codec.New()
+	cryptoAmino.RegisterAmino(cdc)
+
+	var cont tmsecp256k1.PubKeySecp256k1
+	err := cdc.UnmarshalBinaryBare(cryptoPubKey.Bytes(), &cont)
+	if err != nil {
+		return nil, err
+	}
+
+	eeAddress := eeutil.Blake2b256(cont.Bytes())
```

Comment:
eeutil.Blake2b256 will raise panic if fail to create Hash instance. Also, if fail to calculate hash from input, it will ignore errors.

Reply:
Issue raised as #54